frontend/public/components/events.jsx

```diff
@@ -136,43 +139,120 @@ export class EventsList extends React.Component {
+      events.push(
+        <div key="All">
+          <span>
+            <ResourceIcon kind="All" />
+            All resources
+          </span>
+          <EventStream
+            {...this.props}
+            key={'all-resources-event'}
+            type={type}
+            kind={'all'}
+            mock={this.props.mock}
+            textFilter={textFilter}
+          />
+        </div>,
+      );
+    } else {
+      selected.forEach((kind) => {
+        events.push(
+          <div key={kind}>
+            <span>
+              <ResourceIcon kind={kind} />
+              {kind.split('~').pop()}
+            </span>
+            <EventStream
+              {...this.props}
+              key={kind}
+              type={type}
+              kind={kind}
+              mock={this.props.mock}
+              textFilter={textFilter}
+            />
+          </div>,
+        );
+      });
+    }
+    return <div className="co-search co-m-pane__body">{events}</div>;
+  };
+
   render() {
-    const { type, kind, textFilter } = this.state;
-    const { autoFocus = true, mock } = this.props;
+    const { type, selected } = this.state;
+    const { autoFocus = true } = this.props;
 
     return (
       <>
-        <div className="co-m-pane__filter-bar">
-          <div className="co-m-pane__filter-bar-group">
+        <Helmet>
+          <title>Events</title>
+        </Helmet>
+        <PageHeading detail={true} title="Events">
+          <div className="co-search-group">
             <ResourceListDropdown
-              className="btn-group"
-              onChange={(v) => this.setState({ kind: v })}
-              selected={kind}
+              onChange={this.toggleSelected}
+              selected={Array.from(selected)}
               showAll
-              title="All Resources"
+              clearSelection={this.clearSelection}
+              className="co-search-group__resource"
             />
             <Dropdown
-              className="btn-group"
+              className="btn-group co-search-group__resource"
               items={eventTypes}
               onChange={(v) => this.setState({ type: v })}
-              selectedKey={this.state.type}
+              selectedKey={type}
               title="All Types"
             />
-          </div>
-          <div className="co-m-pane__filter-bar-group co-m-pane__filter-bar-group--filter">
             <TextFilter
               autoFocus={autoFocus}
               label="Events by name or message"
               onChange={(e) => this.setState({ textFilter: e.target.value || '' })}
             />
           </div>
-        </div>
-        <EventStream {...this.props} type={type} kind={kind} mock={mock} textFilter={textFilter} />
+          <div className="form-group">
+            <ChipGroup withToolbar defaultIsOpen={false}>
+              <ChipGroupToolbarItem key="resources-category" categoryName="Resource">
+                {[...selected].map((chip) => (
```

Comment:
Why not just...

```suggestion
                {selected.map((chip) => (
```

Reply:
map is not available on Sets, so i have to convert first to array